frappe/core/doctype/navbar_settings/navbar_settings.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.model.document import Document
+from frappe import _
+
+class NavbarSettings(Document):
+	def validate(self):
+		self.validate_standard_navbar_items()
+
+	def validate_standard_navbar_items(self):
+		doc_before_save = self.get_doc_before_save()
+
+		before_save_items = [item for item in \
+			doc_before_save.help_dropdown + doc_before_save.settings_dropdown if item.is_standard]
+
+		after_save_items = [item for item in \
+			self.help_dropdown + self.settings_dropdown if item.is_standard]
+
+		if not frappe.flags.in_patch and (len(before_save_items) > len(after_save_items)):
+			frappe.throw(_("Cannot delete standard navbar items"))
+
+@frappe.whitelist()
+def get_app_logo():
+	app_logo = frappe.db.get_single_value('Navbar Settings', 'app_logo')
+	if not app_logo:
+		app_logo = frappe.get_hooks('app_logo_url')[-1]
+
+	return app_logo
+
+def get_navbar_items():
```

Comment:
why not send the entire `Navbar Settings` document?

Reply:
No specific reason, since this is added in bootinfo just want to keep it lightweight and hence passed only the necessary items. Can send the entire Navbar Settings as well.

Reply:
there is a cost to this too - next time you add a field, you have to remember to update this function too. usually bugs is directly proportional to the lines of code. sending the full doc may be a bit more expensive, but its more robust in my view!

Reply:
send the whole doc!